.github/workflows/build-and-test.yml

```
@@ -0,0 +1,145 @@
+          distribution: 'temurin'
+
+      - name: Suid root tar
+        # Required for the bats libraries cache to restore properly
+        run: sudo chmod +s /bin/tar
+
+      - name: Setup Bats and bats libs
+        id: setup-bats
+        uses: bats-core/bats-action@3.0.0
+
+      - name: Remove suid root tar
+        run: sudo chmod -s /bin/tar
+
+      - uses: sbt/setup-sbt@v1
+
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: Jars
+
+      - name: Clone test projects
+        shell: bash
+        env:
+          BATS_LIB_PATH: ${{ steps.setup-bats.outputs.lib-path }}
+          TERM: xterm
+        working-directory: ./agent
+        run: bin/test_projects
+
+      - uses: actions/setup-java@v4
+        name: Set up Gradle cache
+        with:
+          java-version: ${{ matrix.java }}
+          distribution: 'temurin'
+          cache: gradle
+
+      - uses: actions/setup-java@v4
+        name: Set up Maven cache
+        with:
+          java-version: ${{ matrix.java }}
+          distribution: 'temurin'
+          cache: maven
+
+      - uses: actions/setup-java@v4
+        name: Set up sbt cache
+        with:
+          java-version: ${{ matrix.java }}
+          distribution: 'temurin'
+          cache: sbt
+
+      - name: Build test projects
+        shell: bash
+        env:
+          BATS_LIB_PATH: ${{ steps.setup-bats.outputs.lib-path }}
+          TERM: xterm
+        working-directory: ./agent
+        run: bin/test_install
+
+      - name: Run tests
+        shell: bash
+        env:
+          BATS_LIB_PATH: ${{ steps.setup-bats.outputs.lib-path }}
+          TERM: xterm
+        working-directory: ./agent
+        run: bin/test_run
+
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    needs: test-suite
+    if: ${{ github.ref == 'refs/heads/master' }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+      - uses: actions/download-artifact@v4
+        with:
+          name: Jars
+      - name: Install semantic-release
+        run: |
+          npm i -g \
+              semantic-release \
+              @semantic-release/exec \
+              @semantic-release/git \
+              @semantic-release/changelog \
+              @google/semantic-release-replace-plugin
+      - name: Run semantic-release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.ORG_GRADLE_PROJECT_signingKey }}
+          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.ORG_GRADLE_PROJECT_signingPassword }}
+          ORG_GRADLE_PROJECT_ossrhUsername: ${{ secrets.ORG_GRADLE_PROJECT_ossrhUsername }}
+          ORG_GRADLE_PROJECT_ossrhPassword: ${{ secrets.ORG_GRADLE_PROJECT_ossrhPassword }}
```

GitHub forced these secret identifiers to uppercase. Not sure if it matters.

```suggestion
          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.ORG_GRADLE_PROJECT_SIGNINGKEY }}
          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.ORG_GRADLE_PROJECT_SIGNINGPASSWORD }}
          ORG_GRADLE_PROJECT_ossrhUsername: ${{ secrets.ORG_GRADLE_PROJECT_OSSRHUSERNAME }}
          ORG_GRADLE_PROJECT_ossrhPassword: ${{ secrets.ORG_GRADLE_PROJECT_OSSRHPASSWORD }}
```